BackendApartmentReservation/BackendApartmentReservation/Repositories/TripRepository.cs

```diff
@@ -39,30 +38,40 @@ public async Task<IEnumerable<DbEmployeeAmenitiesChecklist>> GetTripChecklistsWi
 
         public async Task<DbTrip> CreateTrip (CreateTripRequest tripRequest)
         {
+            var tripGroups = new List<DbGroup>();
             var destinationOffice = await _db.Offices.SingleOrDefaultAsync(o => o.ExternalOfficeId == tripRequest.DestinationOfficeId);
             var employees = await _db.Employees
                 .Include(e => e.Office)
                 .Where(e => tripRequest.UserIds.Contains(e.ExternalEmployeeId))
                 .ToListAsync();
 
-            var groups = (from employee in employees
-                          group employee by employee.Office.ExternalOfficeId into g
-                          join office in _db.Offices on g.FirstOrDefault().Office.ExternalOfficeId equals office.ExternalOfficeId
-                          select new DbGroup
-                          {
-                              Employees = g.ToList(),
-                              StartingOffice = g.FirstOrDefault().Office
-                          }).ToList();
+            var employeesGroups = employees.GroupBy(e => e.Office.ExternalOfficeId);
+            foreach (var employeesGroup in employeesGroups)
+            {
+                var tripGroup = new DbGroup
+                {
+                    StartingOffice = await _db.Offices.SingleOrDefaultAsync(o => o.ExternalOfficeId == employeesGroup.Key)
+                };
+                tripGroups.Add(tripGroup);
+                await _db.Groups.AddAsync(tripGroup);
+                
+                var employeeGroups = employeesGroup.Select(e => new DbEmployeeGroup

```

Comment:
Hmm, there is already an API for that in group manager, but only for a single person. We should consider making that a part of group manager API

Reply:
Also, group addition can be moved to manager level. Which makes it:
Create Trip, add Groups, add Checklists